titiler/pgstac/dependencies.py

```diff
@@ -23,6 +24,12 @@
 retry_config = RetrySettings()
 
 
+def db_conn(request: Request) -> Generator[Connection, None, None]:
+    """Return database connection."""
+    with request.app.state.dbpool.connection() as conn:
+        yield conn
```

Comment:
Using a dependency to create the `connection` could enable the reutilization of the connection within the endpoint. e.g., when using the `PGSTACBackend`, before this PR we were creating a new connection on `reader` creation, but with this PR we can initiate the connection at endpoint request and then pass the connection to the reader (which will initiate the cursor)

cc @bitner